pkg/sql/catalog/systemschema/system.go

```diff
@@ -558,27 +560,83 @@ var (
 )
 
 // SystemDatabaseName is the name of the system database.
-const SystemDatabaseName = "system"
+const SystemDatabaseName = catconstants.SystemDatabaseName
 
 // MakeSystemDatabaseDesc constructs a copy of the system database
 // descriptor.
 func MakeSystemDatabaseDesc() catalog.DatabaseDescriptor {
+	priv := privilege.ReadData
 	return dbdesc.NewBuilder(&descpb.DatabaseDescriptor{
 		Name:    SystemDatabaseName,
 		ID:      keys.SystemDatabaseID,
 		Version: 1,
 		// Assign max privileges to root user.
 		Privileges: descpb.NewCustomSuperuserPrivilegeDescriptor(
-			descpb.SystemAllowedPrivileges[keys.SystemDatabaseID], security.NodeUserName()),
+			priv, security.NodeUserName()),
 	}).BuildImmutableDatabase()
 }
 
-func makeTable(desc descpb.TableDescriptor) catalog.TableDescriptor {
+func systemTable(
+	name catconstants.SystemTableName,
+	id descpb.ID,
+	columns []descpb.ColumnDescriptor,
+	families []descpb.ColumnFamilyDescriptor,
+	indexes ...descpb.IndexDescriptor,
+) descpb.TableDescriptor {
+	tbl := descpb.TableDescriptor{
+		Name:                    string(name),
+		ID:                      id,
+		ParentID:                keys.SystemDatabaseID,
+		UnexposedParentSchemaID: keys.PublicSchemaID,
+		Version:                 1,
+		Columns:                 columns,
+		Families:                families,
+		PrimaryIndex:            indexes[0],
+		Indexes:                 indexes[1:],
+		FormatVersion:           descpb.InterleavedFormatVersion,
+		NextMutationID:          1,
+	}
+	for _, col := range columns {
+		if tbl.NextColumnID <= col.ID {
+			tbl.NextColumnID = col.ID + 1
+		}
+	}
+	for _, fam := range families {
+		if tbl.NextFamilyID <= fam.ID {
+			tbl.NextFamilyID = fam.ID + 1
+		}
+	}
+	for _, idx := range indexes {
+		if tbl.NextIndexID <= idx.ID {
+			tbl.NextIndexID = idx.ID + 1
+		}
+	}
+	return tbl
+}
```

Comment:
This function builds a minimum-viable `descpb.TableDescriptor` which can have stuff added to it via the `toCatalogDescriptor` function defined below.